docs/source/examples/VOC2012Segmentation.rst

```diff
@@ -0,0 +1,197 @@
+
+.. literalinclude:: ../../../docs/code/VOC2012Segmentation.py
+   :language: python
+   :start-after: """Organize Dataset / regular import"""
+   :end-at: from tensorbay.label import InstanceMask, SemanticMask
+
+There are already a number of dataloaders in TensorBay SDK provided by the community.
+Thus, instead of writing, importing an available dataloader is also feasible.
+
+.. literalinclude:: ../../../docs/code/VOC2012Segmentation.py
+   :language: python
+   :start-after: """Organize dataset / import dataloader"""
+   :end-before: """"""
+
+.. note::
+
+   Note that catalogs are automatically loaded in available dataloaders, users do not have to write them again.
+
+.. important::
+
+   See :ref:`dataloader table <reference/glossary:dataloader>` for dataloaders with different label types.
+
+*******************
+ Visualize Dataset
+*******************
+
+Optionally, the organized dataset can be visualized by **Pharos**, which is a TensorBay SDK plug-in.
+This step can help users to check whether the dataset is correctly organized.
+Please see :ref:`features/visualization:Visualization` for more details.
+
+****************
+ Upload Dataset
+****************
+
+The organized "VOC2012 Segmentation" dataset can be uploaded to tensorBay for sharing, reuse, etc.
+
+.. literalinclude:: ../../../docs/code/VOC2012Segmentation.py
+   :language: python
+   :start-after: """Upload Dataset"""
+   :end-before: """"""
+
+Similar with Git, the commit step after uploading can record changes to the dataset as a version.
+If needed, do the modifications and commit again.
+Please see :ref:`features/version_control:Version Control` for more details.
+
+**************
+ Read Dataset
+**************
+
+Now "VOC2012 Segmentation" dataset can be read from TensorBay.
+
+.. literalinclude:: ../../../docs/code/VOC2012Segmentation.py
+   :language: python
+   :start-after: """Read Dataset / get dataset"""
+   :end-before: """"""
+
+In :ref:`reference/dataset_structure:Dataset` "VOC2012 Segmentation", there are two
+:ref:`segments <reference/dataset_structure:Segment>`: ``train`` and ``val``.
+Get a segment by passing the required segment name or the index.
+
+.. literalinclude:: ../../../docs/code/VOC2012Segmentation.py
+   :language: python
+   :start-after: """Read Dataset / get segment"""
+   :end-before: """"""
+
+In the ``train`` :ref:`reference/dataset_structure:Segment`,
+there is a sequence of :ref:`reference/dataset_structure:Data`,
+which can be obtained by index.
+
+.. literalinclude:: ../../../docs/code/VOC2012Segmentation.py
+   :language: python
+   :start-after: """Read Dataset / get data"""
+   :end-before: """"""
+
+In each :ref:`reference/dataset_structure:Data`,
+there are one :ref:`reference/label_format/SemanticMask:SemanticMask` annotation and one :ref:`reference/label_format/InstanceMask:InstanceMask` annotation.
+
+.. literalinclude:: ../../../docs/code/VOC2012Segmentation.py
```

Comment:
from PIL import Image, do not need to use try except.
add a blank line between label_semantic_mask and label_instance_mask